File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionOrTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionOrTest extends BaseTest {
    @Test
    void assertOr() {
        var twoOrThree = fromJson("""
                                                   {
                                                   "$$or": [
                                                    { "$$is": "$[0]", "eq": 2 },
                                                    { "$$is": "$[1]", "eq": 3 }
                                                   ]
                                                   }""");
        assertTransformation(fromJson("[2,3]"), twoOrThree, true);
        assertTransformation(fromJson("[1,3]"), twoOrThree, true);
        assertTransformation(fromJson("[2,4]"), twoOrThree, true);
        assertTransformation(fromJson("[1,4]"), twoOrThree, false);
        var firstOrSecond = fromJson("""
            { "$$or": [ "$[0]", "$[1]" ] }""");
        assertTransformation(fromJson("[null,1]"), firstOrSecond, true);
        assertTransformation(fromJson("[null,null]"), firstOrSecond, false);
    }

    @Test
    void inline() {
        assertTransformation(fromJson("[null,0]"), "$$or:$", false);
        assertTransformation(fromJson("[1,0]"), "$$or:$", true);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionOr.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionOr", () => {
  test("assertOr", async () => {
    const twoOrThree = {
      $$or: [
        { $$is: "$[0]", eq: 2 },
        { $$is: "$[1]", eq: 3 },
      ],
    };
    await assertTransformation([2, 3], twoOrThree, true);
    await assertTransformation([1, 3], twoOrThree, true);
    await assertTransformation([2, 4], twoOrThree, true);
    await assertTransformation([1, 4], twoOrThree, false);
    const firstOrSecond = { $$or: ["$[0]", "$[1]"] };
    await assertTransformation([null, 1], firstOrSecond, true);
    await assertTransformation([null, null], firstOrSecond, false);
  });

  test("inline", async () => {
    await assertTransformation([null, 0], "$$or:$", false);
    await assertTransformation([1, 0], "$$or:$", true);
  });
});
